File: 前端代码/score-web/src/api/teacher/types/teacher.ts
export interface TeacherData {
    id?: string
    teacherId: number
    teacherName: string
    gender: string
    title: string
    telephone: string
    email: string
    schoolId?: number
    schoolName: string
    password?: string
}

export interface TableRequestData {
    /** 当前页码 */
    currentPage: number
    /** 查询条数 */
    size: number
    /** 查询参数：关键词 */
    keyword?: string
}

export interface TeacherGroupData {
    schoolName: string
    teacherList: [
        {
            teacherId: number,
            teacherName: string
        }
    ]
}

export type TeacherGroupResponseData = ApiResponseData<{
    records: TeacherGroupData[]
}>

export type TableResponseData = ApiResponseData<{
    records: TeacherData[]
    total: number
}>

File: 后端代码/score-system/src/main/java/com/gxnzd/scoresystem/entity/Teacher.java
package com.gxnzd.scoresystem.entity;

import com.alibaba.excel.annotation.ExcelIgnore;
import com.alibaba.excel.annotation.ExcelProperty;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data   //自动生成所有参数的get()、set()方法
@AllArgsConstructor   //自动生成包含所有参数的构造方法
@NoArgsConstructor    //自动生成无参的构造方法
@TableName("teacher")
public class Teacher {
    @TableId(type = IdType.AUTO)  //声明主键，类型自增
    @ExcelProperty("教师工号")
    private Long teacherId;

    @ExcelProperty("姓名")
    private String teacherName;

    @ExcelProperty("性别")
    private String gender;

    @ExcelProperty("职称")
    private String title;

    @ExcelProperty("电话")
    private String telephone;

    @ExcelProperty("邮箱")
    private String email;

    @ExcelIgnore  // 这个字段会被忽略，不会导出或导入
    private Integer schoolId;

    @TableField(exist = false)
    @ExcelProperty("所属学院")
    private String schoolName;

    @ExcelProperty("密码")
    private String password;
}
